hi-ml-azure/src/health_azure/utils.py

```diff
@@ -463,6 +463,59 @@ def from_string(self, x: str) -> List[str]:
         res = [str(item) for item in x.split(',')]
         return [determine_run_id_type(x) for x in res]
 
+class CheckpointDownloader:
+    def __init__(self, run_id: str, checkpoint_filename: str,  azure_config_json_path: Path = None,
+                 aml_workspace: Workspace = None, download_dir: PathOrString = "checkpoints",
+                 remote_checkpoint_folder: PathOrString = "checkpoints") -> None:
+        """
+        Utility class for downloading checkpoint files from an Azure ML run
+
+       :param run_id: Recovery ID of the run from which to load the checkpoint.
+        :param checkpoint_filename: Name of the checkpoint file, expected to be inside the
+        `outputs/checkpoints/` directory (e.g. `"best_checkpoint.ckpt"`).
+        :param azure_config_json_path: An optional Azure ML settings (JSON file) to use to access the specified
+            experiment run. If not running inside an AML Run, and no aml_workspace object is provided, this
+            is required.
+        :param aml_workspace: An optional Azure ML Workspace object. If not running inside an AML Run, and no
+            azure_config_json_path is provided, this is required.
+        :param download_dir: The local directory in which to save the downloaded checkpoint files.
+        :param remote_checkpoint_folder: The remote folder from which to download the checkpoint file
+        """
+        self.azure_config_json_path = azure_config_json_path
+        self.aml_workspace = aml_workspace
+        self.run_id = run_id
+        self.checkpoint_filename = checkpoint_filename
+        self.download_dir = Path(download_dir)
+        self.remote_checkpoint_folder = remote_checkpoint_folder
+
+    @property
+    def local_checkpoint_path(self) -> Path:
+        return self.download_dir / self.run_recovery_id.split(":")[1] / self.checkpoint_filename
```

Comment:
run_id above should be run_recovery_id

Reply:
Yes good catch, I meant to change this one to run_recovery too.
@vale-salvatelli  - I changed it to run_id because it doesn't need to be a run recovery id (this is behaviour from an older Azure sdk), but that said, you can still pass in a run recovery id and it will work